pypesto/objective/base.py

```diff
@@ -178,6 +183,8 @@ def __call__(
         x_full = self.pre_post_processor.preprocess(x=x)
 
         # compute result
+        if self.share_return_dict:
```

Comment:
why is this necessary?

Reply:
Otherwise there might be some argument error, because `return_dict` is not in every custom user `objective.call_unprocessed` signature. But I'll switch to Daniel's suggestion now